### PR TITLE
Upgrade pip in PyPI and Test PyPI workflow

### PR DESCRIPTION
**Pull request by toshihikoyanase**
## Motivation

Related to the issue reported in https://github.com/optuna/optuna/pull/2431#issuecomment-821919625.
The pip seems to fail to resolve the appropriate version of `twine`. This may be due to the old pip.

## Description of the changes

Upgrade pip to the latest version.

### CODE REVIEW

**Comment by toshihikoyanase** [comment]
@himkt @HideakiImamura Could you review this PR, please?

File: .github/workflows/pypi-publish.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@ jobs:
 
     - name: Install twine
       run: >-
+        python -m pip install --upgrade pip
```
Comment (nzw0301) [inline .github/workflows/pypi-publish.yml:25]: [really nitpick]: the next line uses `install -U`, so how about unifying them for consistency?

**Comment by toshihikoyanase** [comment]
@nzw0301 Thank you for your suggestion. I applied it. PTAL.

**Review by nzw0301 — APPROVED** [review]
I'm not one of the assignees, but let me approve this PR. LGTM.

**Pull request merged**